FileSystem/FileOpeningBuilder.cs

```diff
@@ -79,6 +79,11 @@ public Stream Open(string path)
 
         private void HandleNotNativelySupportedConfigurations(FileInfo path)
         {
+            if (_truncate && _createNew && path.Exists)
```

Comment:
```suggestion
            if (_createNew && path.Exists)
```

Reply:
The only purpose of that function is to handle configurations not supported natively.
`createNew` on it's own is supported natively.

Reply:
Isn't the whole point of your PR to standardize the way `CreateNew` behaves with certain configs? This takes explicit care of that. Move it to another method if you want

Reply:
The advantage of leaving the handling of createNew to the OS is that it avoids race conditions because the OS can do it atomically.

Reply:
We are simply throwing early in a guaranteed fashion, I fail to see how race conditions are more of a threat when making this check more restrictive then otherwise.